server/game/StatusToken.js

```diff
@@ -1,12 +1,47 @@
-const GameObject = require('./GameObject');
+const EffectSource = require('./EffectSource');
+const AbilityDsl = require('./abilitydsl');
+const { Players, Durations } = require('./Constants');
 
-class StatusToken extends GameObject {
+class StatusToken extends EffectSource {
     constructor(game, card, isHonored) {
         super(game, isHonored ? 'Honored Token' : 'Dishonored Token');
         this.honored = !!isHonored;
         this.dishonored = !isHonored;
         this.card = card;
         this.type = 'token';
+        this.persistentEffects = [];
+
+        this.applyHonorEffects();
+    }
+
+    applyHonorEffects() {
```

Comment:
My instinct is that this is in the wrong place, and that it should be dealt with in either basecard or drawcard with the rest of the effect housekeeping.  It looks like you're messing with effects outside of `checkGameState` and that's going to mess up how state-based effects work.....

Reply:
I'm sure you are right, I will see what I can do.  I originally wanted the "source" of the effect to be the token, I suppose I can still do that via the drawcard though :)